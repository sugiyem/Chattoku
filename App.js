import { StatusBar } from "expo-status-bar";
import React from "react";
import * as Notifications from "expo-notifications";
import { LogBox } from "react-native";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";

import DashboardScreen from "./src/screens/DashboardScreen";
import ForgotPasswordScreen from "./src/screens/ForgotPasswordScreen";
import LoginScreen from "./src/screens/LoginScreen";
import SignupScreen from "./src/screens/SignupScreen";

const Stack = createNativeStackNavigator();

// Ignore the console warning from EventEmitter.removeListener
// Warning most likely from using react-native-gifted-chat
LogBox.ignoreLogs(["EventEmitter.removeListener"]);

Notifications.setNotificationHandler({
  handleNotification: async () => ({
    shouldShowAlert: true,
    shouldPlaySound: false,
    shouldSetBadge: false
  })
});

export default function App() {
  return (
    <NavigationContainer>
      <StatusBar style="light" />
      <Stack.Navigator
        initialRouteName="Login"
        screenOptions={{ headerShown: false }}
      >
        <Stack.Screen name="Login" component={LoginScreen} />
        <Stack.Screen name="Signup" component={SignupScreen} />
        <Stack.Screen name="ForgotPassword" component={ForgotPasswordScreen} />
        <Stack.Screen name="Dashboard" component={DashboardScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
