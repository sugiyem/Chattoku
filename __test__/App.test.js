import React from "react";
import renderer from "react-test-renderer";
import App from "../App";

jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");
jest.mock("../src/services/Firebase/Config", () => ({
  firebase: {
    auth: () => ({
      onAuthStateChanged: jest.fn()
    })
  }
}));

describe("<App />", () => {
  it("Renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
