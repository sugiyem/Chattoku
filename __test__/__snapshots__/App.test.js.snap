// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Login"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4287225076,
                4285745896,
              ],
              "endPoint": Array [
                1,
                1,
              ],
              "locations": undefined,
              "startPoint": Array [
                0,
                0,
              ],
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RCTSafeAreaView
            emulateUnlessSupported={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "paddingBottom": 5,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                  "paddingTop": 5,
                  "rowGap": 15,
                },
              ]
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "borderRadius": 1000,
                    "height": 80,
                    "width": 250,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignSelf": "stretch",
                    "borderRadius": 10,
                    "marginBottom": 0,
                    "marginLeft": 20,
                    "marginRight": 20,
                    "marginTop": 0,
                    "paddingBottom": 20,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 20,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#483d8b",
                      "fontSize": 25,
                      "fontWeight": "600",
                      "textAlign": "center",
                    },
                  ]
                }
              >
                 Log In 
              </Text>
              <View
                style={
                  Object {
                    "margin": 5,
                    "paddingTop": 20,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "black",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    collapsable={false}
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "gray",
                        "fontSize": 16,
                        "left": 5,
                        "opacity": 1,
                        "padding": 0,
                        "position": "absolute",
                        "transform": Array [
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scaleX": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                        "zIndex": 10,
                      }
                    }
                  >
                    Email
                  </Text>
                  <Text
                    collapsable={false}
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "navy",
                        "fontSize": 16,
                        "left": 5,
                        "opacity": 0,
                        "padding": 0,
                        "position": "absolute",
                        "transform": Array [
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scaleX": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                        "zIndex": 10,
                      }
                    }
                  >
                    Email
                  </Text>
                  <TextInput
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    secureTextEntry={false}
                    style={
                      Object {
                        "borderRadius": 10,
                        "fontSize": 16,
                        "padding": 7,
                        "paddingLeft": 5,
                        "width": "100%",
                        "zIndex": 20,
                      }
                    }
                    value=""
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "margin": 5,
                    "paddingTop": 20,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "black",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    collapsable={false}
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "gray",
                        "fontSize": 16,
                        "left": 5,
                        "opacity": 1,
                        "padding": 0,
                        "position": "absolute",
                        "transform": Array [
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scaleX": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                        "zIndex": 10,
                      }
                    }
                  >
                    Password
                  </Text>
                  <Text
                    collapsable={false}
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "navy",
                        "fontSize": 16,
                        "left": 5,
                        "opacity": 0,
                        "padding": 0,
                        "position": "absolute",
                        "transform": Array [
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scaleX": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                        "zIndex": 10,
                      }
                    }
                  >
                    Password
                  </Text>
                  <TextInput
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    secureTextEntry={true}
                    style={
                      Object {
                        "borderRadius": 10,
                        "fontSize": 16,
                        "padding": 7,
                        "paddingLeft": 5,
                        "width": "100%",
                        "zIndex": 20,
                      }
                    }
                    value=""
                  />
                </View>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "navy",
                    "borderColor": "#44d0fe",
                    "borderRadius": 15,
                    "borderWidth": 0.8,
                    "marginBottom": 10,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 10,
                    "opacity": 1,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "color": "#44d0fe",
                        "fontSize": 18,
                        "fontWeight": "600",
                      },
                    ]
                  }
                >
                  Log In
                </Text>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#44d0fe",
                    "borderColor": "navy",
                    "borderRadius": 15,
                    "borderWidth": 0.8,
                    "marginBottom": 10,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 10,
                    "opacity": 1,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "color": "navy",
                        "fontSize": 16,
                      },
                    ]
                  }
                >
                  Don't Have an Account? Sign Up Here
                </Text>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#44d0fe",
                    "borderColor": "navy",
                    "borderRadius": 15,
                    "borderWidth": 0.8,
                    "marginBottom": 10,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 10,
                    "opacity": 1,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  }
                }
              >
                <Text
                  color="#ffffff"
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "color": "navy",
                        "fontSize": 16,
                      },
                    ]
                  }
                >
                  Forgot Your Password? Click Here
                </Text>
              </View>
            </View>
          </RCTSafeAreaView>
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
